common/src/main/java/io/netty/util/concurrent/AbstractScheduledEventExecutor.java

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.util.concurrent;
+
+import java.util.Iterator;
+import java.util.PriorityQueue;
+import java.util.Queue;
+import java.util.concurrent.Callable;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Abstract base class for {@link EventExecutor}s that want to support scheduling.
+ */
+public abstract class AbstractScheduledEventExecutor extends AbstractEventExecutor {
+
+    Queue<ScheduledFutureTask<?>> scheduledTaskQueue;
```

Comment:
Assigning a member variable into a local variable usually means copying the address of the member variable to a CPU register.  Once we know the address is stored in the register, we do not need to copy again. i.e. use the address value in the register which is much faster.

When the compiler decided not to optimize enough, it is likely that we end up with accessing the address of the member variable via memory rather than via a register.

So, it's a kind of hint to the compiler that we don't need to access it from memory over and over again. This is especially useful when the member variable is volatile.

Of course, it really depends on what compiler chooses to do. If the method it is optimizing is complicated enough and it decided to store the address in the stack memory, there might not be any gain, for example. 

Nevertheless, I think it is safer to move the frequently-used value to a local variable, because it should never perform worse unless the compiler does a really weird job.